### PR TITLE
VGS: Remove Asserts from GBPing that are causing crashes

### PR DESCRIPTION
**Pull request by pluddy**
Asserts make sense in testing, but not in production.  These asserts are causing our top crashes in Sideline.  #SAD

### CODE REVIEW

File: GBPing/GBPing.m [hunk 1]
```diff
@@ -366,7 +366,7 @@ -(void)listenOnce {
     enum { kBufferSize = 65535 };
     
     buffer = malloc(kBufferSize);
-    assert(buffer);
+    if (buffer == nil) { return; }
```
Comment (pluddy) [inline GBPing/GBPing.m]: wondering if i should set `err` to `errno` here?

**Review by jhofker — APPROVED** [review]
Looks good. Do we want to use `NSAssert` and then catch on the other side, or does that make it too complex still?

File: GBPing/GBPing.m [hunk 2]
```diff
@@ -388,7 +388,7 @@ -(void)listenOnce {
             NSMutableData *packet;
 
             packet = [NSMutableData dataWithBytes:buffer length:(NSUInteger) bytesRead];
-            assert(packet);
+            if (packet == nil) { return; }
```
Comment (pluddy) [inline GBPing/GBPing.m]: similar here, do I set `pingSummary.status` to `GBPingStatusFail` ?

File: GBPing/GBPing.m [hunk 3]
```diff
@@ -498,7 +498,7 @@ -(void)sendPing {
                 packet = [self pingPacketWithType:kICMPv6TypeEchoRequest payload:payload requiresChecksum:NO];
             } break;
             default: {
-                assert(NO);
+                return;
```
Comment (pluddy) [inline GBPing/GBPing.m]: same here

**Comment by pluddy** [comment]
hmmm, `NSAssert` is actually not a bad idea... then we would be assured that this changed logic doesnt effect the way GBPing works...  Idk, tough call

**Comment by jhofker** [comment]
I'd say yes to setting a failed state in those conditions. The ping is definitely failing, so the correct block should get called.

**Comment by ReesKlintworth** [comment]
I think the way this is implemented makes sense. If we're hitting these failures, that means that the ping is failing. We shouldn't have to worry about catching or dealing with that on our side, something is still wrong with the device.

File: GBPing/GBPing.m (hunk 2 above)
Comment (pluddy) [reply GBPing/GBPing.m]: actually... I need the packet to create `pingSummary`.  ill probably just return here as this should be caught above I think anyway

**Pull request merged**